### PR TITLE
Fix towncrier bug and pin version

### PR DESCRIPTION
**Pull request by Cadair**
Towncrier changed their toml API, which they are allowed to do as it's private, so also hard pin the version so we don't have this happen again.

### CODE REVIEW

File: baldrick/plugins/github_towncrier_changelog.py [hunk 1]
```diff
@@ -8,46 +8,7 @@
...
-            'directory': config.get('directory'),
-            'sections': sections,
-            'types': types,
-            'template': config.get('template', _template_fname),
-            'start_line': config.get('start_string', _start_string),
-            'title_format': config.get('title_format', _title_format),
-            'issue_format': config.get('issue_format'),
-            'underlines': config.get('underlines', _underlines)
-        }
+from towncrier._settings import parse_toml as parse_towncrier_toml
```
Comment (pllim) [inline baldrick/plugins/github_towncrier_changelog.py:11]: If only it is not in a "hidden" module...

**Review by pllim — APPROVED** [review]
Whatever works...

**Comment by pllim** [comment]
What is this?

`pytest.PytestDeprecationWarning: <class 'pytest_flake8.Flake8Item'> is not using a cooperative constructor and only takes {'fspath', 'parent'}.`

**Comment by pllim** [comment]
Do we really need `pytest-flake8` ?

**Comment by Cadair** [comment]
> Do we really need `pytest-flake8` ?

No?

**Comment by pllim** [comment]
Failures look real:

```
DEBUG    urllib3.connectionpool:connectionpool.py:1005 Starting new HTTPS connection (1): api.github.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:465 [https://api.github.com:443](https://api.github.com/) "GET /repos/nota/repo HTTP/1.1" 404 112
```

**Comment by pllim** [comment]
Maybe the default branch thing doesn't work.

```
    @property
    def default_branch(self):
>       return self.repo_info["default_branch"]
E       KeyError: 'default_branch'
```

**Comment by Cadair** [comment]
I think our mock responses are out of date. I don't really have the time to fix them atm.

**Comment by pllim** [comment]
Well, if this patch works for you, feel free to merge then.

**Pull request merged**